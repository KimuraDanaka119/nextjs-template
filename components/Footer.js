import React from 'react';

const Footer = () => (
  <div className="footer">
    <div className="container">
      This is footer
    </div>
  </div>
);

export default Footer;
