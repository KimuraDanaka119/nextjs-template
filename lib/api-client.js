import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
