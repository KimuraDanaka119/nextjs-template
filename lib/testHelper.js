import ApiClient from './api-client';

function t(key) {
  return key;
}

const config = {
  env: 'test',
  versionHash: 'test',
  paths: {
    static: '/static/test',
    styles: '/static/test/styles',
    images: '/static/test/images',
    locales: '/static/test/locales',
  },
  api: {
    baseURL: '.',
  },
};

const apiClient = new ApiClient('.');

export { t, config, apiClient };
