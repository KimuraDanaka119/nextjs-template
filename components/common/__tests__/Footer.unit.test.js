import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { undecorated as Footer } from '../Footer';
import { t } from '../../../lib/testHelper';

describe('<Footer />', () => {
  it('should render correctly', () => {
    const footer = shallow(<Footer t={t}/>);
    expect(toJson(footer)).toMatchSnapshot();
  });
});
