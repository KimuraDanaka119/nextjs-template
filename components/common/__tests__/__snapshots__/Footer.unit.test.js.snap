// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<div
  className="footer"
>
  <div
    className="container"
  >
    common:footer.text
  </div>
</div>
`;
